OfficeDev/microsoft-teams-library-js
# Add react router for navigation testing

## Description

This PR adds React Router to enable navigation between tabs when caching is enabled. There is no Nav component because tabs are populated in the appDefinition files. The onResume function is changed to navigate based on the URL provided.

> Summarize the changes, including the goals and reasons for this change. Be sure to call out any technical or behavior changes that reviewers should be aware of.

> If this Pull Request should close/resolve any issues when merged, use [the special syntax for that](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) here.

### Main changes in the PR:

1. add react router to the package
2. add an empty page for testing
3. enable routing in the app.tsx file
4. update webpack config so the second page can be loaded by url

### Validation performed:
- change the URL to https://localhost:4000/second-page
- ensure the content is an empty page
- switch tabs with caching enabled in a test host and ensure the onResume function properly navigates to the empty page.


### Unit Tests added:

> Unit tests are required for all changes. If no unit tests were added as part of this change, please explain why they aren't necessary.

<Yes/No>

### End-to-end tests added:

<Yes/No>

## Additional Requirements

### Change file added:

> Ensure the change file meets the [formatting requirements](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md#change-log-using-beachball).

<Yes/No>

### Related PRs:

> Remove this section if n/a

### Next/remaining steps:

> List the next or remaining steps in implementing the overall feature in subsequent PRs (or is the feature 100% complete after this?).

> Remove this section if n/a

- [ ] Item 1
- [ ] Item 2

### Screenshots:

> Remove this section if n/a

| Before     | After      |
| ---------- | ---------- |
| < image1 > | < image2 > |
